hkmc2/shared/src/test/mlscript-compile/Option.mls

```diff
@@ -6,7 +6,7 @@ open Predef
 // TODO
 // type Option[A] = Option.Some(A) | Option.None
 
-module Option with ...
+object Option with ...
```

Comment:
Why?

Reply:
Some function used to have a signature that returns an `Option`, but `Option` used to be module, so it requires users to write `fun f: module Option` which I guess is not what we want... I don't know why no function returns an option now. Maybe we removed them in some following commits.

I just realized that this is wrong and it should be a part of definition overloading, i.e. there should be a type called `Option` that is different from the module `Option`. As it doesn't cause error now, I'll change it back to an module.